src/algorithms/geometry/graham_scan.js

```diff
@@ -0,0 +1,63 @@
+const vectorOp = require('./vector_operations2d.js');
+
+/**
+ * Given an array P with N points on a two dimensional space,
+ * the function grahamScan calculates the convex hull of P
+ * in O(N * log(N)) time and using O(N) space.
+ * Implemented algorithm: Graham's Scan.
+ *
+ * @param An array P with N points, each point should be a literal,
+ * example: {x:0,y:0}.
+ * @return An array P' with N' points which belongs to the convex hull of P.
+ *
+ * Note: If exists a subset S of P that contains only collinear points
+ * which are present on the convex hull of P, then only the pair with
+ * the largest distance will be present on P'.
+ */
+const grahamScan = function(P) {
+  if (P.length <= 3) {
+    return P;
+  }
+  preprocessing(P);
+  console.log(P);
+  const convexHull = [P[0], P[1]];
+  for (let i = 2; i < P.length; i++) {
+    let j = convexHull.length;
+    while (j >= 2 &&
+            vectorOp.
+              parallelogramArea(convexHull[j-2], convexHull[j-1], P[i]) <= 0) {
```

Comment:
Can we use the `isClockwise` helper here? (or `!isCounterClockwise`)

Reply:
Actually we can't, because both isClockwise and isCounterClockwise return false when the points are collinear, but we want to remove them from the convexHull when they are collinear.

Reply:
Actually we can use !isCounterClockwise, it will do the same job. I'll change it soon.